### PR TITLE
[IMP] redux-storage: support v4.1.2

### PR DESCRIPTION
**Pull request by alexey-pelykh**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/react-stack/redux-storage/commit/cbe98031ffea10b6ab882f840cd5e900c46473c4
- [x] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

If removing a declaration:
- [ ] If a package was never on DefinitelyTyped, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

File: types/redux-storage/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
     "private": true,
     "dependencies": {
-        "redux": "^3.6.0"
+        "redux": "^3.6.0 || ^4.0.0"
```
Comment (PranavSenthilnathan) [inline types/redux-storage/package.json]: Is there a reason not to just take 4.0.0? I'm assuming this condition does the same thing.

**Review by PranavSenthilnathan — CHANGES_REQUESTED** [review]

File: types/redux-storage/package.json (hunk 1 above)
Comment (alexey-pelykh) [reply types/redux-storage/package.json]: They actually do different things: `^3.6.0` mean "3.x.x" and `^4.0.0` mean `4.x.x` - please check at https://semver.npmjs.com/

**Comment by alexey-pelykh** [comment]
@PranavSenthilnathan your comments have been addressed - please review.

**Comment by alexey-pelykh** [comment]
Reopened as #35197

**Pull request closed without merging**